types/detox/index.d.ts

```diff
@@ -8,6 +8,13 @@ declare global {
     const device: Detox.Device;
     const element: Detox.Element;
     const waitFor: Detox.WaitFor;
+    /**
+     * `dexpect` ("Detox-expect") intentionally has the same type as `expect`.
+     * `expect` conflicts with the global `expect` of jest and therefore
+     * exporting it doesn't work which is why `dexpect` was introduced. `expect`
+     * is kept for backwards compatibility.
+     */
+    const dexpect: Detox.Expect<Detox.Expect<any>>;
```

Comment:
I would recommend moving `dexpect` outside of the `declare global` block.
That way, it won't be exposed globally, but it will still be available for you to export on line 444 as you do currently. As you mentioned, this doesn't resolve the issue on the `const expect` line, but it sounds like that's the best you're aiming for right now.

Reply:
Thanks for pointing that out @uniqueiniquity. It was just wrong with the global `dexpect` because there is no such global in the actual detox js. So I've moved it outside of the `declare global`. I've also changed the jsdoc comment to be an ordinary comment because otherwise it is shown on mouse over.